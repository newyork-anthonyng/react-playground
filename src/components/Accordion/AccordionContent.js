import React from 'react';
import T from 'prop-types';

const AccordionContent = ({
  id,
  ariaLabelledBy,
  isOpen,
  children,
}) => {
  if (!isOpen) return null;

  return (
    <dd
      role="region"
      aria-labelledBy={ariaLabelledBy}
    >
      {children}
    </dd>

  );
};

AccordionContent.defaultProps = {
  isOpen: false,
};

AccordionContent.propTypes = {
  ariaLabelledBy: T.string.isRequired,
  id: T.string.isRequired,
  isOpen: T.bool.isRequired,
  children: T.node.isRequired,
};

export default AccordionContent;
