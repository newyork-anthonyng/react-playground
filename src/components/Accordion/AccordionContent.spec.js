import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import AccordionContent from './AccordionContent';

const defaultProps = {
  ariaLabelledBy: 'some-element-id',
  id: 'my-element-id',
  isOpen: true,
};

it('should render correctly when open', () => {
  const wrapper = shallow(
    <AccordionContent
      {...defaultProps}
      isOpen
    >
      <h1>I am content</h1>
    </AccordionContent>
  );

  expect(toJSON(wrapper)).toMatchSnapshot();
});

it('should render correctly when not open', () => {
  const wrapper = shallow(
    <AccordionContent
      {...defaultProps}
      isOpen={false}
    >
      <h1>I am content</h1>
    </AccordionContent>
  );

  expect(toJSON(wrapper)).toMatchSnapshot();
});
