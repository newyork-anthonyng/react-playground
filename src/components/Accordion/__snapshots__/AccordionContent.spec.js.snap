// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly when not open 1`] = `null`;

exports[`should render correctly when open 1`] = `
<dd
  aria-labelledBy="some-element-id"
  role="region"
>
  <h1>
    I am content
  </h1>
</dd>
`;
